controller/appcontroller.go

```diff
@@ -284,9 +290,17 @@ func (projCache *appProjCache) GetAppProject(ctx context.Context) (*appv1.AppPro
 	return projCache.appProj, nil
 }
 
+// getAppProj gets the AppProject for the given Application app.
 func (ctrl *ApplicationController) getAppProj(app *appv1.Application) (*appv1.AppProject, error) {
 	projCache, _ := ctrl.projByNameCache.LoadOrStore(app.Spec.GetProject(), ctrl.newAppProjCache(app.Spec.GetProject()))
-	return projCache.(*appProjCache).GetAppProject(context.TODO())
+	proj, err := projCache.(*appProjCache).GetAppProject(context.TODO())
+	if err != nil {
+		return nil, err
```

Comment:
Should this error be wrapped for easier debugging?

Reply:
Yeah, I actually agree this is a good idea :+1: